test/cases/ostree-vsphere.sh

```diff
@@ -0,0 +1,787 @@
+greenprint "📋 Copying vmdk image to vsphere datacenter 6.7"
+govc datastore.mkdir -dc="Datacenter6.7" -ds="datastore-225" "${COMPOSE_ID}-image"
+govc datastore.cp -dc="Datacenter7.0-AMD" -ds="datastore-21" -dc-target="Datacenter6.7" -ds-target="datastore-225" "${COMPOSE_ID}-image/${ISO_FILENAME}" "${COMPOSE_ID}-image/${ISO_FILENAME}"
+
+##################################################################
+##
+## Create vm on datacenter7.0-amd and test it
+##
+##################################################################
+# Create vm with vmdk image
+greenprint "📋 Create vm in vsphere datacenter 7.0-AMD"
+DC70_VSPHERE_VM_NAME="${COMPOSE_ID}-70"
+govc vm.create -dc="Datacenter7.0-AMD" -ds="datastore-21" -pool="/Datacenter7.0-AMD/host/Cluster7.0-AMD/Resources" \
+    -net="VM Network" -net.adapter=vmxnet3 -disk.controller=pvscsi -on=false -c=2 -m=4096 \
+    -g="${GUEST_ID_DC70}" -firmware=efi "${DC70_VSPHERE_VM_NAME}"
+govc vm.disk.attach -ds="datastore-21" -dc="Datacenter7.0-AMD" -vm "${DC70_VSPHERE_VM_NAME}" \
+    -link=false -disk="${COMPOSE_ID}-image/${ISO_FILENAME}"
+govc vm.power -on -dc="Datacenter7.0-AMD" "${DC70_VSPHERE_VM_NAME}"
+DC70_GUEST_ADDRESS=$(govc vm.ip -dc=Datacenter7.0-AMD -wait=10m "${DC70_VSPHERE_VM_NAME}")
+greenprint "🛃 Edge VM IP address is: ${DC70_GUEST_ADDRESS}"
+
+# Run ansible check on edge vm
+# Check for ssh ready to go.
+greenprint "🛃 Checking for SSH is ready to go"
+for _ in $(seq 0 30); do
+    RESULTS="$(wait_for_ssh_up "${DC70_GUEST_ADDRESS}")"
+    if [[ $RESULTS == 1 ]]; then
+        echo "SSH is ready now! 🥳"
+        break
+    fi
+    sleep 10
+done
+
+# Check image installation result
+check_result
+
+greenprint "🕹 Get ostree install commit value"
+INSTALL_HASH=$(curl "${PROD_REPO_URL}/refs/heads/${OSTREE_REF}")
+
+sudo tee "${TEMPDIR}"/inventory > /dev/null << EOF
+[ostree_guest]
+${DC70_GUEST_ADDRESS}
+
+[ostree_guest:vars]
+ansible_python_interpreter=/usr/bin/python3
+ansible_user=admin
+ansible_private_key_file=${SSH_KEY}
+ansible_ssh_common_args="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+ansible_become=yes
+ansible_become_method=sudo
+ansible_become_pass=${IGNITION_USER_PASSWORD}
+EOF
+
+# Test IoT/Edge OS
+sudo ansible-playbook -v -i "${TEMPDIR}"/inventory \
+    -e ignition="true" \
+    -e image_type=redhat \
+    -e ostree_commit="${INSTALL_HASH}" \
+    -e skip_rollback_test="false" \
+    -e edge_type=edge-vsphere \
+    -e fdo_credential="false" \
+    -e sysroot_ro="$SYSROOT_RO" \
+    /usr/share/tests/osbuild-composer/ansible/check_ostree.yaml || RESULTS=0
+check_result
+
+##################################################################
+##
+## Create vm on datacenter6.7 and test it
+##
+##################################################################
+# Create vm with vmdk image
+greenprint "📋 Create vm in vsphere datacenter 6.7"
+DC67_VSPHERE_VM_NAME="${COMPOSE_ID}-67"
```

Comment:
Please move this definition at the start of this script.

Reply:
This variable uses {COMPOSE_ID}, if move it to start of script, the COMPOSE_ID is empty.